package-managers/nuget.md

```diff
@@ -86,21 +94,67 @@ After making sure our feed is a public one we can consume it in Rider. To work w
+
 #### Private feed
 
 We will need the URL for the feed we want to connect to. Learn more about [determining the feed URL](#determining-the-feed-url).
 
+To register our credentials in the `NuGet.config` file we use the following commands:
+
+<p class="alert alert-info">
+    <strong>Note:</strong> These credentials are non-transferable.
+</p>
+
+```shell
+nuget setapikey [apikey] -source [feedUrl]
+nuget sources add -name [name] -source [feedUrl] -user [username] -pass [pwd]
+```
+
+To update an already registered package source:
+
+```shell
+nuget setapikey [apikey] -source [feedUrl]
+nuget sources update -name [name] -source [feedUrl] -user [username] -pass [pwd]
+```
+
+If we want dont want to add our credentials to the global `NuGet.config` but to a specific one, we use the `-configFile` parameter and specify the path to our prefered `NuGet.config` file:
+
+```shell
+nuget setapikey [apikey] -source [feedUrl] -configFile [configFilePath]
+nuget sources add -name [name] -source [feedUrl] -user [username] -pass [pwd] -configFile [path]
+```
+
+If we want our credentials to be transferable for lets say, use by a build server we add the `-StorePasswordInClearText` flag:
+
+```shell
+nuget setapikey [apikey] -source [feedUrl] -configFile [configFilePath]
+nuget sources update -name [name] -source [feedUrl] -user [username] -pass [pwd] -configFile [path] -StorePasswordInClearText
+```
+
 #### Public feed
 
 We will need the URL for the feed we want to connect to. Learn more about [determining the feed URL](#determining-the-feed-url).
 
+To add our feed to the global `NuGet.config` file we can use the command:
+
+```shell
+nuget sources add -name [name] -source [feedUrl]
+```
+
+If our build server needs to know about this feed, we can add it to a `NuGet.config` file in our project and push it to our code repository. To configure this feed in the specific `NuGet.config` file, we add the  `-configFile` parameter:
+
+```shell
+nuget sources add -name [name] -source [feedUrl] -configFile [path]
+```
+
 ### dotnet CLI
 
 In this section we are going to add our packages using the dotnet CLI. Make sure to [install the .NET Core tooling](https://www.microsoft.com/net/core). The folks (and docs) at Microsoft can help you on your way.
 
 #### Private feed
 
-We will need the URL for the feed we want to connect to. Learn more about [determining the feed URL](#determining-the-feed-url).
+We will need the URL for the feed we want to connect to. Learn more about [determining the feed URL](#determining-the-feed-url). For installation per package, we need the pre-authenticated feed URL. As we do not recommend this option because of security reasons we do recommend using the [NuGet CLI](#nuget-cli) to configure your feeds instead.

```

Comment:
"For installation per package, we need the pre-authenticated feed URL."

Not sure I get this one. Install is always per package? Think that first sentence can be removed.

Maybe:
`Alternatively, we can use the pre-authenticated feed URL. Since pre-authenticated feed URLs contain your personal access token in the URL, we advise using the [NuGet command-line tool (CLI)](#nuget-cli) to configure your feeds instead. If you do want to make use of a pre-authenticated endpoint, we recommend creating a separate access token specifically for this purpose so that it can be revoked at any time. Access tokens can be managed from your MyGet profile.`